AK/AsyncStreamBuffer.h

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2024, Dan Klishch <danilklishch@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Coroutine.h>
+#include <AK/Error.h>
+
+namespace AK {
+
+class AsyncStreamBuffer {
+    AK_MAKE_NONCOPYABLE(AsyncStreamBuffer);
+
+public:
+    AsyncStreamBuffer()
+    {
+        m_capacity = min_capacity;
+        m_data = (u8*)kmalloc(m_capacity);
```

Comment:
No C-style casts please. :^)

Reply:
It wasn't me, it was copilot, I swear :)